lib/active_record/connection_adapters/percona_adapter.rb

```diff
@@ -142,6 +156,14 @@ def execute_and_free(sql, name = nil)
       private
 
       attr_reader :mysql_adapter, :logger, :runner, :cli_generator
+
+      # Checks whether the sql statement is an ALTER TABLE
+      #
+      # @param sql [String]
+      # @return [Boolean]
+      def alter_statement?(sql)
+        sql =~ /alter table/i
```

Comment:
it's not the exact same regex and it'd introduce an undesirable coupling between the adapter and the `AlterArgument`